lib/user.rb

```diff
@@ -0,0 +1,29 @@
+require 'pry'
+
+
+class User
+
+attr_reader :name, :jokes
+
+  def initialize(name)
+    @name = name
+    @jokes = []
+  end
+
+  def learn(joke)
+    @jokes << joke
+  end
+
+  def tell(who, what)
+    who.jokes << what
+  end
+
+  def joke_by_id(number)
```

Comment:
You don't need `self` here (although it doesn't technically hurt anything, it's just not very ruby-like).  You can just call `@jokes`, since you're still inside the user class at this point. 

This is a great job of getting this to work with each! You could also have done `find`, since that enumerable is specifically meant for this purpose. 

```
@jokes.find do |joke|
     joke.id == number 
end
```